inference-engine/src/vpu/graph_transformer/src/stages/interpolate.cpp

```diff
@@ -25,92 +27,89 @@ void FrontEnd::parseInterpolate(const Model& model, const ie::CNNLayerPtr& _laye
+
+    const auto modeIsSupported = cmp(interpolateMode, g_nearest) || cmp(interpolateMode, g_linear) || cmp(interpolateMode, g_linear_onnx);
+    VPU_THROW_UNLESS(modeIsSupported, "Current Interpolate supports 'nearest' and 'linear' modes only, actual {}", interpolateMode);
+
+    const auto paramIsSupported = ic == oc && in == 1 && on == 1 && isPadZeros(padsBegin) && isPadZeros(padsEnd);
+    VPU_THROW_UNLESS(paramIsSupported, "Current Interpolate does not support paddings, batches, and resize by channels");
+
+    if (cmp(interpolateMode, g_nearest)) {
+        // current "Resample" supports the following "Interpolate" modes only:
+        // coordinate_transformation_mode = half_pixel; nearest_mode = round_prefer_ceil;
+        // coordinate_transformation_mode = asymmetric; nearest_mode = floor;
+        // other "Interpolate" modes are translated to the default ones
+        const auto anti = _layer->GetParamAsBool(g_antialias, false);
+        const auto coordinateTransformation = _layer->GetParamAsString(g_coordinate_transformation_mode, g_half_pixel);
+        const auto near = _layer->GetParamAsString(g_nearest_mode, g_round_prefer_floor);
+        auto coordinateTransformationMode = InterpolateCoordTransMode::HalfPixel;
+        auto nearestMode = InterpolateNearestMode::RoundPreferCeil;
+
+        const auto coordModeIt   = coordTransformModeMap.find(coordinateTransformation);
+        const auto nearestModeIt = nearestModeMap.find(near);
+
+        VPU_THROW_UNLESS(coordModeIt != coordTransformModeMap.end(), "Interpolate stage does not support this coordinate transforation mode");
+        VPU_THROW_UNLESS(nearestModeIt != nearestModeMap.end(), "Interpolate stage does not support this nearest transforation mode");
+        coordinateTransformationMode = coordModeIt->second;
+        nearestMode = nearestModeIt->second;
+
+        _stageBuilder->addResampleNearestStage(model,
+                                                _layer->name,
+                                                _layer,
+                                                anti,
+                                                coordinateTransformationMode,
+                                                nearestMode,
+                                                -1.0f,
+                                                input,
+                                                output);
+    } else if (cmp(interpolateMode, g_linear) || cmp(interpolateMode, g_linear_onnx)) {
+        // current "Interp" supports modes "align_corners" and "asymmetric" only
+        // other "Interpolate" modes are translated to the default ones
+        const auto coordinateTransformation = _layer->GetParamAsString(g_coordinate_transformation_mode, g_half_pixel);
+        auto coordinateTransformationMode = InterpolateCoordTransMode::HalfPixel;
+        auto mode = InterpolateMode::Linear;
```

Comment:
unused default value

Reply:
`I mean, you do not use these default values anymore, you overwrite it with coordModeIt->second; at line 103`

Well, yes. But in case when these values are not specified, then they won't be overwritten. Then we'll need the default ones. Right?

Reply:
Although, think i got your point)

Reply:
Well, you get layer parameter as string or default 
`const auto interpolateMode = _layer->GetParamAsString(g_mode, g_nearest);`
then, you do the string parameter to enum value mapping and throw exception if no such string is in keys
`const auto interpolateModeIt = interpModeMap.find(interpolateMode);`
`VPU_THROW_UNLESS(interpolateModeIt != interpModeMap.end(), ...`
and then you write the enum value got from map in the mode variable
`mode = interpolateModeIt->second;`

In summary, it is an impossible case 
> But in case when these values are not specified

Reply:
yes, right. I got it. Thanks